doc: revamp kata containers getting started guide (#12144)
cilium-team/cilium: Documentation/gettingstarted/kata.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _kata:

***************************
Kata Containers with Cilium
***************************

`Kata Containers <https://katacontainers.io/>`_ is an open source project that
provides a secure container runtime with lightweight virtual machines that feel
and perform like containers, but provide stronger workload isolation using
hardware virtualization technology as a second layer of defense.

> Similar to the OCI runtime ``runc`` provided by Docker, Cilium can be used with

Reviewer: ```suggestion
Kata Containers implements OCI runtime spec, just like ``runc`` that is used by Docker. Cilium can
```

> Kata Containers, providing a higher degree of security at the network layer and

Reviewer: ```suggestion
be used along with Kata Containers, using both enables higher degree of security.
```

> at the compute layer with Kata.

Reviewer: ```suggestion
 Kata Containers enhances security in the compute layer, while Cilium provides policy and observability in networking layer.```

> This guide provides a walkthrough of installing Cilium on Kata Containers. It

Reviewer: ```suggestion
This guide shows how to install Cilium along with Kata Containers. It
```
...since "provides" was used in previous paragraph.

there are two supported ways called CRI-O and CRI-containerd. It is up to you
to choose the one that you want, but you have to pick one.

Refer to the section :ref:`k8s_requirements` for detailed instruction on how to
prepare your Kubernetes environment and make sure to use Kubernetes >= 1.12.
Then, follow the
`official guide to run Kata Containers with Kubernetes <https://github.com/kata-containers/documentation/blob/master/how-to/run-kata-with-k8s.md>`_.

.. note::
   Minimum version of kubernetes 1.12 is required to use the RuntimeClass Feature
   for Kata Container runtime described below.

With your Kubernetes cluster ready, you can now proceed to deploy Cilium.

Deploy Cilium
=============

.. include:: k8s-install-download-release.rst

Deploy Cilium release via Helm:

  .. tabs::

     .. group-tab:: Using CRI-O

        .. parsed-literal::

           helm install cilium |CHART_RELEASE| \\
             --namespace kube-system \\
             --set global.containerRuntime.integration=crio

     .. group-tab:: Using CRI-containerd

        .. parsed-literal::

           helm install cilium |CHART_RELEASE| \\
             --namespace kube-system \\

>              --set global.containerRuntime.integration=containerd

Reviewer: I am not sure if `global.containerRuntime.integration` is really needed. I used Cilium with containerd and containerd+kata without setting the parameter, and didn't see any issues.

Author: I am not sure either but it was part of the documentation and the distinction for CRI-O and containerd was clearly emphasized.

Reviewer: Ah, so actually `crio` is the only special value. In practice, I believe `containerd` basically doesn't have special needs.
```
$ git grep .Values.global.containerRuntime.integration  install/kubernetes 
install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:{{- if not (eq .Values.global.containerRuntime.integration "crio") }}
install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:{{- if not (eq .Values.global.containerRuntime.integration "crio") }}
install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:{{- if not (eq .Values.global.containerRuntime.integration "crio") }}
$
```

.. include:: k8s-install-validate.rst

> Install Kata on a running Kubernetes Cluster

Reviewer: I'm not sure whether this section is really helpful, especially in the context of Cilium. To me this guide just needs to carry the message that Cilium works with Kata, and you can install it quite simply using the helm chart, other information is beyond the scope.

Author: I agree. I actually intended to update that section but didn't have time yet but I'm also OK to remove it entirely.
